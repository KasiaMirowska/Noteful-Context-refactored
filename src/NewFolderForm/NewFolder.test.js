import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NewFolder from './NewFolderForm';


describe('NewFolder comsponent', () => {
    const props = {
        match: {
            params: {
                path: '/newFolder'
            }
        }
    }
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NewFolder {...props}/>, div)
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', () => {
        const newFolder = renderer.create(<NewFolder {...props}/>);
        expect(newFolder.toJSON()).toMatchSnapshot();
    });
});
