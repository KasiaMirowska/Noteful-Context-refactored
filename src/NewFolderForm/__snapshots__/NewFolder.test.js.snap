// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFolder comsponent renders UI as expected 1`] = `
<div
  className="note-list"
>
  <div
    className="noteList-header-container"
  >
    <h2
      className="noteList-header"
    >
      Create a folder
    </h2>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-fields"
    >
      <label
        htmlFor="folder-name"
      >
        Name
      </label>
      <input
        aria-label="folder-name"
        aria-required="true"
        id="name"
        name="name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="button-field"
    >
      <button
        type="submit"
      >
        Add Folder
      </button>
    </div>
  </form>
</div>
`;
